priam/src/main/java/com/netflix/priam/defaultimpl/PriamGuiceModule.java

```diff
@@ -148,4 +158,17 @@ HostAndPort providePort(PriamConfiguration configuration) {
             throw new IllegalStateException("Did not find a valid HttpConnector for the server", ex);
         }
     }
+
+    @Provides
+    @Singleton
+    @SDBCredentialProvider
+    AWSCredentialsProvider provideAWSCredentialsProvider(AWSCredentialsProvider defaultAWSCredentialsProvider, @Named("awsRoleAssumptionARN") String awsRoleAssumptionARN) {
+        if (Strings.isNullOrEmpty(awsRoleAssumptionARN)) {
+            return defaultAWSCredentialsProvider;
+        } else {
+            return new STSAssumeRoleSessionCredentialsProvider(defaultAWSCredentialsProvider, awsRoleAssumptionARN,
+                    Joiner.on('-').join("AwsRoleAssumptionSession", priamConfiguration.getCassandraConfiguration().getClusterName(),
+                            priamConfiguration.getCassandraConfiguration().getDataCenterSuffix(), DateTime.now(DateTimeZone.UTC).toString()));
```

Comment:
From the AWS documentation concerning the role session name:

> The regex used to validate this parameter is a string of characters consisting of upper- and lower-case alphanumeric characters with no spaces. You can also include underscores or any of the following characters: =,.@-

Maybe use a custom `DateTime` formatter or just a long timestamp to avoid violating this restriction.